assets/ts/stop/components/DeparturesWithBadge.tsx

```diff
@@ -57,16 +48,23 @@ const DeparturesWithBadge = ({
   if (!(priorityBadge || secondaryBadge)) return <>{children}</>;
   const displayBadge = (priorityBadge || secondaryBadge)!;
 
+  // if the priority badge is present and there is only one departure, show the badge
+  // this is because we check for predictions and return a single time list if there are none
```

Comment:
A single time list is also always returned if there are commuter rail predictions (as it's designed to only display one time at time). 

So. I'm not sure that tying _this_ PR's logic to "time list happens to have length of 1" makes all that much sense.